Sources/UIWindow.swift

```diff
@@ -47,16 +47,23 @@ public class UIWindow: UIView {
             }
 
         case .ended:
-            currentTouch.runTouchActionOnRecognizerHierachy { $0.touchesEnded(allTouches, with: event) }
             if !currentTouch.hasBeenCancelledByAGestureRecognizer {
                 hitView.touchesEnded(allTouches, with: event)
             }
 
+            currentTouch.runTouchActionOnRecognizerHierachy { $0.touchesEnded(allTouches, with: event) }
```

Comment:
this change i'm not so sure about (compared to the touchesBegan one) – it may be ok like this, but my gut feeling is that it was more correct how it was

Reply:
i can keep the old order, we just have to compute `hasBeenCancelledByAGestureRecognizer` before running the recognizer chain